### DISCUSSION TITLE
Don't works on server, I don't know what else to try

### DISCUSSION THREAD

**Discussion by LeandroVCastro** [geckosio/geckos.io#138] [Help / Questions] [answered]
I'm developing a game using geckos.io.
Initially, I was developing using socket.io, but I was has problems with latency and lag so I chose to try geckos.io.

The migration was very easy and on my localhost this works perfectly.
But when I run it on the server, just doesn't connect.

I'm using EC2 AWS. 
I already enabled UDP ports on the firewall in AWS.
I'm using nginx to server my application.

This is the link: https://app.dinobattle.one/

What's happened? When I click on "play" button should start the connect, but here start the error.
There's no error exactly, just don't connect. 
My server doesn't receive the "onConnect" event, so the game doesn't work.
The server even doesn't print any message or error.

I have tested everything I can think.

Anyone could help me, please?

Thanks.

**Comment by yandeu** [comment, marked as answer]
Make sure the UDP traffic does not go through nginx. The clients need a direct UDP connection to your EC2 instance.

**Reply by LeandroVCastro** [reply]
How could I do it?

```
server {

        root /var/www/api.dinobattle.one/html;
        index index.html index.htm index.nginx-debian.html;

        server_name api.dinobattle.one www.api.dinobattle.one;

        location / {
              proxy_pass http://localhost:3000;
              proxy_http_version 1.1;
              proxy_set_header Upgrade $http_upgrade;
              proxy_set_header Connection 'upgrade';
              proxy_set_header Host $host;
              proxy_cache_bypass $http_upgrade;
        }

    listen [::]:443 ssl ipv6only=on; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/api.dinobattle.one/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/api.dinobattle.one/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}
server {
    if ($host = api.dinobattle.one) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


        listen 80;
        listen [::]:80;

        server_name api.dinobattle.one www.api.dinobattle.one;
    return 404; # managed by Certbot


}
```

This is my nginx config

**Reply by yandeu** [reply]
Just make sure the client can directly connect to the gecko's server ip via UDP.

All wrtc connections are a direct connection between geckos and the webbrowser.

If you host your game on the same server as nginx, and its ip is 1.2.3.4, make sure the browser can connect to 1.2.3.4/upd.

**Reply by LeandroVCastro** [reply]
I will need to set the port on the client?
I don't know if I got exactly what you said.

For example, I'm using geckos with express on nodejs and serving it on port 3000. So I will need to set port = 3000 on client side?

**Reply by yandeu** [reply]
Well that's not what I meant. But yes the client library needs the same port.

Besides 3000/TCP you need to open 1024-65530/UPD. The signaling is done on port 3000. After signaling, each player connects directly on a random udp port.

**Reply by LeandroVCastro** [reply]
![image](https://user-images.githubusercontent.com/11198617/143311296-fa475341-5b93-42a1-b1ae-eab2f0707213.png)


This is my firewall config.
I think it's okay.

Not?

**Reply by LeandroVCastro** [reply]
I did some changes
I deleted my EC2 instance so I created a new in São Paulo (I'm from Brazil).
So I just copy the nginx configs etc and it works perfectly.
I don't know exactly what solves this issue, but it's solved.
I suppose it's because of distance or something... Make sense?
Anyway this topic can be marked as "solved".

**Reply by yandeu** [reply]
I don't think it's because of the distance.

**Reply by LeandroVCastro** [reply]
I don't think so. Was just a supposition.
Anyway, thanks for the help. I'm so happy to use this lib. It's very useful.
